### ISSUE TITLE
Virtual Packages

### ISSUE THREAD

**Issue by rmadsen-ks** [opentap/opentap#493]
Virtual Packages are packages which can be implicitly installed by running a .exe or msi installer. The package contains a pattern and if the pattern is met, it means that the package is installed. 

Hence a package can be installed without technically downloading or installing any files.

This could be useful for
 - Dotnet 6 Runtime
 - Python Runtime
 - Keysight IO Libraries
 - ...

**Comment by bdirenzo-keysight** [comment]
@rmadsen-ks how is this different than just ActionStep? Is it that the file pattern/registry pattern can also check if the package exists? (even if not installed by a .TapPackage?)

**Comment by alnlarsen** [comment]
I like this feature a lot, but it looks like it will have drastically different behavior from non-virtual packages. In the example package we ship `Packages/Dotnet6/dotnet-runtime-installer.exe`. I assume this means we will want to download the package definition first and check the patterns.

If we add this I think we should also be able to verify installations. E.g. 

`ActionStep ActionName="Verify" ExeFile="Packages/Dotnet6/dotnet-runtime-installer.exe" Arguments="VerifyInstall"`

This is just an example, I don't know if there is such an option. It could be that we should consider a different approach from an ActionStep since it would mean the user can't delete installers from virtual packages.

**Comment by rmadsen-ks** [comment]
@bdirenzo-keysight, the only difference is that you don't need to actually download and install it, if the content of the package is already installed. Also, the package could be registered (when install is not needed) in a user-local directory, so elevation is not necessesary. 

@alnlarsen, what is the purpose of VerifyInstall? The installation is verified by the pattern matching. Maybe e.g dotnet.exe could be called to further verify, but maybe that should actually be another type of pattern. "ProcessOutputPattern".

**Comment by alnlarsen** [comment]
@rmadsen-ks yes, something like 'ProcessOutputPattern' could work too. The purpose of VerifyInstall is that a dotnet installation is not necessarily working just because 'dotnet.exe' exists. I just think virtual packages should have the option to check if they work because we can't verify that their files are correctly installed like with normal packages.

**Comment by rmadsen-ks** [comment]
We wont have time for this in 9.18.

**Comment by rmadsen-ks** [comment]
Another problem this could fix is if the application has been manually deleted, but the package is still present in the package cache. The pattern could tell us that this package is not actually installed although the package.xml is located in the right place.
